### PR TITLE
Use the project.lock.json to determine the compatible framework for projects

### PR DESCRIPTION
**Pull request by davidfowl**
This removes the last set of nuget code that the runtime uses to determine compatibility. All compatibility moves to restore and the runtime just reads the lock file to build a model. This is the final part of #2622

/cc @anurse

### CODE REVIEW

File: src/Microsoft.Dnx.Tooling/ProjectExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
```
Comment (BrennanConroy) [inline src/Microsoft.Dnx.Tooling/ProjectExtensions.cs]: copyright

Comment (davidfowl) [inline src/Microsoft.Dnx.Tooling/ProjectExtensions.cs]: Do we have a script for this?

File: src/Microsoft.Dnx.Runtime/Project.cs [hunk 2]
```diff
@@ -169,19 +169,12 @@ public static bool TryGetProject(string path, out Project project, ICollection<D
...
             }
 
-            IEnumerable<TargetFrameworkInformation> compatibleConfigurations;
-            if (VersionUtility.GetNearest(targetFramework, GetTargetFrameworks(), out compatibleConfigurations) &&
-                compatibleConfigurations.Any())
-            {
-                targetFrameworkInfo = compatibleConfigurations.FirstOrDefault();
-            }
-
             return targetFrameworkInfo ?? _defaultTargetFrameworkConfiguration;
```
Comment (davidfowl) [inline src/Microsoft.Dnx.Runtime/Project.cs]: Why not?

Comment (BrennanConroy) [inline src/Microsoft.Dnx.Runtime/Project.cs]: Nevermind

File: src/Microsoft.Dnx.Tooling/ProjectExtensions.cs (hunk 1 above)
Comment (BrennanConroy) [inline src/Microsoft.Dnx.Tooling/ProjectExtensions.cs]: Not that I know of

**Comment by BrennanConroy** [comment]
Squash, copyright and :shipit:

**Comment by davidfowl** [comment]
This is merged. Feel free to keep reviewing.

**Pull request closed without merging**